### PR TITLE
Update to rustls 0.22.0 alpha.6

### PR DESCRIPTION
**Pull request by tottoto**
Updates to `rustls` 0.22.0-alpha.6, fixes a feature requirement, and adds a ci job to check each feature.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -145,3 +145,17 @@ jobs:
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+      - name: Install cargo-hack
+        uses: taiki-e/install-action@cargo-hack
+      - name: Check each features
+        run: cargo hack --no-dev-deps check --each-feature
```
Comment (cpu) [inline .github/workflows/build.yml:161]: WDYT about using `--feature-powerset ` instead of `--each-feature`? Would it inflate CI time too substantially?

**Review by cpu — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (tottoto) [reply .github/workflows/build.yml]: Updates to use `--feature-powerset` to test the CI time.

Comment (tottoto) [reply .github/workflows/build.yml]: In this test, the CI time increased from `32s` (https://github.com/tottoto/hyper-rustls/actions/runs/7050201648/job/19190355371) to `44s` (https://github.com/rustls/hyper-rustls/actions/runs/7050273761/job/19190655897). I think it is within an acceptable range as the impact on the overall CI is small.

Comment (cpu) [reply .github/workflows/build.yml]: Agreed, that seems totally reasonable to me. Would you mind squashing the commit to use the powerset into the commit adding the job?

Comment (tottoto) [reply .github/workflows/build.yml]: Squashed the two commits.

Comment (cpu) [reply .github/workflows/build.yml]: Thanks!

**Comment by djc** [comment]
Thanks!

**Pull request merged**